ocaml/xcp-rrdd/test/transport/test_common.ml

```diff
@@ -97,37 +105,52 @@ let print_type = function
   | Rrd.Gauge ->
       "Gauge"
 
-let print_value_type = function
+let print_value = function
   | Rrd.VT_Float x ->
       "Float " ^ string_of_float x
   | Rrd.VT_Int64 x ->
-      "Float " ^ Int64.to_string x
+      "Int64 " ^ Int64.to_string x
   | Rrd.VT_Unknown ->
       "Unknown"
 
-let assert_ds_equal (owner1, ds1) (owner2, ds2) =
-  assert_equal ~printer:print_owner owner1 owner2 ;
+let value e = Alcotest.testable (Fmt.of_to_string print_value) (equal_value e)
+
+let assert_ds_equal d1 d2 =
   let open Ds in
-  assert_equal ~printer:print_string ds1.ds_name ds2.ds_name ;
-  assert_equal ~printer:print_string ds1.ds_description ds2.ds_description ;
-  assert_equal ~cmp:are_value_types_equal ~printer:print_value_type ds1.ds_value
-    ds2.ds_value ;
-  assert_equal ~printer:print_type ds1.ds_type ds2.ds_type ;
-  assert_equal ~printer:string_of_bool ds1.ds_default ds2.ds_default ;
-  assert_equal ~printer:string_of_float ds1.ds_min ds2.ds_min ;
-  assert_equal ~printer:string_of_float ds1.ds_max ds2.ds_max ;
-  assert_equal ~printer:print_string ds1.ds_units ds2.ds_units
+  Alcotest.(check string) "Names match" d1.ds_name d2.ds_name ;
+  Alcotest.(check string)
+    "Descriptions match" d1.ds_description d2.ds_description ;
+  Alcotest.check (value 0.01) "Values match" d1.ds_value d2.ds_value ;
+  assert (d1.ds_type = d2.ds_type) ;
+  Alcotest.(check bool) "Defaults match" d1.ds_default d2.ds_default ;
+  compare_float "Minimums match" d1.ds_min d2.ds_min ;
+  compare_float "Maximums match" d1.ds_max d2.ds_max ;
+  Alcotest.(check string) "Units match" d1.ds_units d2.ds_units
+
+let assert_ds_equal (owner1, ds1) (owner2, ds2) =
+  Alcotest.check owner "Owners match" owner1 owner2 ;
+  assert_ds_equal ds1 ds2
 
 let assert_payloads_equal payload1 payload2 =
-  Rrd_protocol.(
-    assert_equal ~msg:"Incorrect timestamp read" ~printer:Int64.to_string
-      payload1.timestamp payload2.timestamp ;
-    assert_equal ~msg:"Incorrect number of datasources read"
-      ~printer:string_of_int
-      (List.length payload1.datasources)
-      (List.length payload2.datasources) ;
-    List.iter2 assert_ds_equal payload1.datasources payload2.datasources
-  )
+  let open Rrd_protocol in
+  Alcotest.(check int64)
+    "Timestamps match" payload1.timestamp payload2.timestamp ;
+  Alcotest.(check int)
+    "Number of datasources read matches written ones"
+    (List.length payload1.datasources)
+    (List.length payload2.datasources) ;
+  List.iter2 assert_ds_equal payload1.datasources payload2.datasources
 
-let make_shared_file () =
-  Filename.temp_file ~temp_dir:"/dev/shm" "test-metrics" ".tmp"
+let make_shared_file ?(k = 0) () =
+  Filename.temp_file ~temp_dir:"/dev/shm"
```

Comment:
If this is causing test failures it might be because there are leftovers from a previous test, or we run multiple tests in parallel and they clash on the names. Filename.temp_file tries 1000 times to generate a name (using only 6 random hex digits though).

We could try to make this more unique by including the PID perhaps? And maybe register an `at_exit` handler to safe unlink them, so even if the test fails/gets interrupted we'd clean it up.

Reply:
We could also increment a (global) number ourselves to make this more unique. I see we generate 4096 files in a test, so we could include PID + globally incrementing number in the suffix.

Reply:
There could be a conflict between test_scale and test_unit when run simultaneously. Adding the number like I did would remove the possibility of conflict. We could add the name of the executable, or pid (as you say), or even function. But I don't see the point with the current solution